### ISSUE TITLE
Enforce parsing of all files

### ISSUE THREAD

**Issue by dbogatov** [James-Yu/LaTeX-Workshop#1153]
**Is your feature request related to a problem? Please describe.**

I dynamically substitute files, so I need parser (and chktex specifically) to work with all `*.tex` files.

**Describe the solution you'd like**

Have an option to lint all files, not just the root and what's included from the root.

**Additional context**

Here is the project structure:

```
main.tex
preamble.tex
cli.tex
assignments/
    hw-1/
        content.tex
        meta.tex
    hw-2/
        content.tex
        meta.tex
    ...
```

and these lines of code in main

```latex
\input{assignments/\assignment/meta} % chktex 27
\begin{document}
    \input{assignments/\assignment/content} % chktex 27
\end{document}
```

as you may guess, value of `\assignment` is externally changed.
This way I can compile my 20 assignments with script, reusing same preamble.

Because the content is not included explicitly (eg. ` \input{assignments/hw-1/content}`), chktex does not work for it.

**Comment by jlelong** [comment]
I am wondering whether this is the good way to proceed.
Maybe, it would be better to support macro replacement for expressions of the type
```
\\(def|(re)?newcommand){\\commandname}{value}
```
when computing the tree structure of the LaTeX project.

**Comment by jlelong** [comment]
Parsing `\newcommand` definition may not be satisfying as we will never be able to correctly evaluate the `\if` section and recursivity may kill us. 
I am actually thinking of adding a new configuration setting to define some macros (without arguments) in the same spirit as the `defines` entry in `c_cpp_properties.json`, see [here](https://code.visualstudio.com/docs/languages/cpp#_configuring-intellisense). 

Any thoughts on that?

**Comment by dbogatov** [comment]
Allowing to define macros in settings would be nice.

My concern, however, was not about parsing my own macros.
I want a setting that instructs the extension to parse all `*.tex` files, not only those which a reachable in a tree starting at main.

**Comment by jlelong** [comment]
> I want a setting that instructs the extension to parse all `*.tex` files, not only those which a reachable in a tree starting at main.

Unfortunately, this is unlikely to be implemented as it does not fit in the current logic of the extension.

**Comment by dbogatov** [comment]
Replying to https://github.com/James-Yu/LaTeX-Workshop/issues/731#issuecomment-468950368

For me a big win would be at least having chktex parsing all the files.

**Comment by jlelong** [comment]
Closed by PR #1199.
